### PR TITLE
Introduce key/value REST test transform

### PR DESCRIPTION
**Pull request by jakelandis**
This commit introduces `replaceValueTextByKeyValue` which allows 
replacements based on a given key/value pair. This is useful 
for REST test transforms that that may not have a well known structure.

This commit also implements this new transform to fix a Watcher compat test.
This commit also fixes a latent bug that would prevent the traversal when multiple
differing transforms matched.

### CODE REVIEW

**Comment by jakelandis** [comment]
@elasticmachine update branch

File: x-pack/plugin/watcher/qa/rest/build.gradle [hunk 1]
```diff
@@ -48,11 +48,13 @@ tasks.named("transformV7RestTests").configure({ task ->
   task.replaceKeyInDo("watcher.stop", "xpack-watcher.stop")
 
   task.addAllowedWarningRegex(".*_xpack/watcher/.* is deprecated.*")
+  task.addAllowedWarningRegex("\\[types removal\\].*")
+  task.replaceValueTextByKeyValue("path", "/my_index/my_type/{{ctx.watch_id}}", "/my_index/_doc/{{ctx.watch_id}}")
```
Comment (pgomulka) [inline x-pack/plugin/watcher/qa/rest/build.gradle:52]: how about we add this with a testname or a comment indicateing which test is the transformation affecting?

**Review by pgomulka — APPROVED** [review]
LGTM - I left one question

File: x-pack/plugin/watcher/qa/rest/build.gradle (hunk 1 above)
Comment (jakelandis) [reply x-pack/plugin/watcher/qa/rest/build.gradle]: done on [`68b6e3e` (#75193)](https://github.com/elastic/elasticsearch/pull/75193/commits/68b6e3e3b9aad74c2cab3976aeea582b3971a2f2)

**Pull request merged**